scripts/prepare_int_data.py

```diff
@@ -0,0 +1,185 @@
+    pass
+
+def ExitProgram(message):
+  print(message, file=sys.stderr)
+  os._exit(1)
+
+def RunCommand(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  if os.system(command) != 0:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+
+def GetCommandStdout(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  try:
+    output = subprocess.check_output(command, shell = True)
+  except:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+  return output
+
+def GetNames(script_dir, text_dir, int_dir):
+  command = "{0}/internal/get_names.py {1} > {2}/names".format(
+              script_dir, text_dir, int_dir) 
+  RunCommand(command)
+
+def GetNumTrainSets(int_dir):
+  command = "cat {0}/names | wc -l".format(int_dir)
+  line = subprocess.check_output(command, shell = True) 
+  try:
+    a = line.split()
+    assert len(a) == 1
+    ans = int(a[0])
+  except:
+    sys.exit("prepare_int_data.py: error: unexpected output '{0}' from command {1}".format(
+                line, command))
+  return ans
+
+def CopyFile(src, dest):
+  try:
+    shutil.copyfile(src, dest)
+  except: 
+    sys.exit("prepare_int_data.py: error copying {0} to {1}".format(src, dest))
+
+# returns num-words in vocab.
+def GetNumWords(vocab):
+  command = "tail -n 1 {0}".format(vocab)
+  line = subprocess.check_output(command, shell = True)
+  try:
+    a = line.split()
+    assert len(a) == 2
+    ans = int(a[1])
+  except:
+    sys.exit("prepare_int_data.py: error: unexpected output '{0}' from command {1}".format(
+                line, command))
+  return ans
+
+# we can include the preparation of the dev data in the following
+# by adding "dev dev" to the contents of int_dir/names
+def PrepareData(line):
+  [int, name] = line.split()
+  if os.path.exists(args.text_dir + "/" + name + ".txt.gz"):
+    command = "set -o pipefail; gunzip -c {text_dir}/{name}.txt.gz | text_to_int.py {vocab} | gzip -c > {int_dir}/{int}.txt.gz 2>{int_dir}/log/{int}.log".format(
+                  text_dir = args.text_dir, name = name, vocab = args.vocab, int_dir = args.int_dir, int = int)
+    output = GetCommandStdout(command)
+  else:
+    command = "set -o pipefail; cat {text_dir}/{name}.txt | text_to_int.py {vocab} | gzip -c > {int_dir}/{int}.txt.gz 2>{int_dir}/log/{int}.log".format(
+                  text_dir = args.text_dir, name = name, vocab = args.vocab, int_dir = args.int_dir, int = int)
+    output = GetCommandStdout(command)
+
+# return the names of train and dev datasets 
+scriptdir = os.path.abspath(os.path.dirname(sys.argv[0]))
+GetNames(scriptdir, args.text_dir, args.int_dir)
+
+# copy vocab to the output directory 
+src = args.vocab 
+dest = args.int_dir + "/words.txt"
+CopyFile(src, dest)
+
+# return the number of train sets 
+with open(args.int_dir + "/num_train_sets", "w") as file:
+  file.write(str(GetNumTrainSets(args.int_dir)))
+
+# return num-words in vocab 
```

Comment:
again, nothing is returning.
And here and above, it's good practice to close your file handles.